File: src/main/java/swtech/pageDesignControl/entity/PersonInfomation.java
package swtech.pageDesignControl.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import lombok.Data;
import lombok.experimental.Accessors;

/**
 * <p>
 *
 * </p>
 *
 * @author 袁君选
 * @since 2019-11-28
 */
@Data
@Accessors(chain = true)
public class PersonInfomation {

    private static final long serialVersionUID = 1L;

    /**
     * 部门：
     */
    private String pdepartment;

    /**
     * 岗位：
     */
    private String pposition;

    /**
     * 填表日期:
     */
    private String pcreatedate;

    /**
     * 姓   名
     */
    private String pname;

    /**
     * 性    别
     */
    private String psex;

    /**
     * 民     族
     */
    private String pnation;

    /**
     * 婚姻状况
     */
    private String pmarriage;

    /**
     * 政治面貌
     */
    private String ppolitics;

    /**
     * 出生年月
     */
    private String pbirthday;

    /**
     * 最高学历
     */
    private String pacademic;

    /**
     * 专  业
     */
    private String pmajor;

    /**
     * 毕业学校
     */
    private String pschool;

    /**
     * 毕业时间
     */
    private String pgraduationdate;

    /**
     * 学  位
     */
    private String pdegree;

    /**
     * 职   称
     */
    private String pprofessional;

    /**
     * 照片
     */
    private String pimg;

    /**
     * 电话号码
     */
    private String ptelephonenumber;

    /**
     * 入职时间
     */
    private String phiredate;

    /**
     * 身份证地址
     */
    private String paddress;

    /**
     * 身份证号码
     */
    private String pidcard;

    /**
     * 银行卡账号
     */
    private String pbankaccount;

    /**
     * 银行卡开户行
     */
    private String popenbank;

    /**
     * 教育经历起止时间
     */
    private String peducationdate1;

    private String peducationdate2;

    private String peducationdate3;

    /**
     * 教育经历学校
     */
    private String peducationschool1;

    private String peducationschool2;

    private String peducationschool3;

    /**
     * 教育经历专业
     */
    private String peducationmajor1;

    private String peducationmajor2;

    private String peducationmajor3;

    /**
     * 教育经历所获学位
     */
    private String peducationdegree1;

    private String peducationdegree2;

    private String peducationdegree3;

    /**
     * 教育经历所获证书
     */
    private String peducationcertificate1;

    private String peducationcertificate2;

    private String peducationcertificate3;

    /**
     * 专业资格证书
     */
    private String pqualification;

    /**
     * 获取资格时间
     */
    private String pqualificationdate;

    /**
     * cad
     */
    private String pprofessionalcad;

    /**
     * ps
     */
    private String pprofessionalps;

    /**
     * 3dmax
     */
    private String pprofessional3dmax;

    /**
     * 天正建筑
     */
    private String pprofessionaltarch;

    /**
     * 职业技能其他
     */
    private String pprofessionalother;

    /**
     * 工作经历起止时间
     */
    private String pexperiencedate1;

    private String pexperiencedate2;

    private String pexperiencedate3;

    /**
     * 工作经历起止时间
     */
    private String pexperienceunit1;

    private String pexperienceunit2;

    private String pexperienceunit3;

    /**
     * 工作经历工作岗位
     */
    private String pexperiencepost1;

    private String pexperiencepost2;

    private String pexperiencepost3;

    /**
     * 工作经历薪资水平
     */
    private String pexperiencepay1;

    private String pexperiencepay2;

    private String pexperiencepay3;

    /**
     * 工作经历离职原因
     */
    private String pexperiencedimission1;

    private String pexperiencedimission2;

    private String pexperiencedimission3;

    /**
     * 工作经历证明人及电话
     */
    private String pexperiencedimissionnumber1;

    private String pexperiencedimissionnumber2;

    private String pexperiencedimissionnumber3;

    /**
     * 家庭成员姓名
     */
    private String pfamilyfamilyname1;

    private String pfamilyfamilyname2;

    /**
     * 家庭成员年龄
     */
    private String pfamilyfamilyage1;

    private String pfamilyfamilyage2;

    /**
     * 家庭成员工作单位
     */
    private String pfamilyfamilyunit1;

    private String pfamilyfamilyunit2;

    /**
     * 家庭成员职位
     */
    private String pfamilyfamilypost1;

    private String pfamilyfamilypost2;

    /**
     * 家庭成员联系电话
     */
    private String pfamilyfamilynumber1;

    private String pfamilyfamilynumber2;

    /**
     * 应急联系人
     */
    private String pemergencycontactname;

    /**
     * 联系电话
     */
    private String pemergencycontactnumber;

    /**
     * 向公司提供的资料身份证
     */
    private String pinformationid;

    /**
     * 向公司提供的资料银行卡
     */
    private String pinformationbank;

    /**
     * 向公司提供的资料学历
     */
    private String pinformationeducation;

    /**
     * 向公司提供的资料其他
     */
    private String pinformationother;

    /**
     * 是否有离职证明文件
     */
    private String pemploymentseparation;

    /**
     * 是否有犯罪记录
     */
    private String pcriminalrecord;

    /**
     * 自增id
     */
    @TableId(value = "pid", type = IdType.AUTO)
    private Integer pid;

    /**
     * 用户id
     */
    private Integer uid;


}

File: src/main/resources/static/myjs/personInfomation.js
/**
 * 个人档案start
 */
$(function () {

    var str = window.location.href;
    var uid = str.substring(str.indexOf("=")+1);
    alert(uid)
    getPersonInfomation(uid);
})

/**
 * 获取单个个人档案
 */
function getPersonInfomation(uid) {
    $.ajax({
        url: pageDesignControl_HOST + '/manage/personinfomation/selectbyuid',
        type: 'Get',
        contentType: 'application/json',
        dataType: 'json',
        data: {"uid": uid},
        async: false,
        success: function (msg) {
            if (msg.status == 200) {
                $("#img_area").html('<img id="pimg" src="'+msg.msg.pimg+'"/>') ;
                $(".personInfomationManageEdit #pname").val(msg.msg.pname);
                $(".personInfomationManageEdit #uid").val(msg.msg.uid);
                $(".personInfomationManageEdit #pdepartment").val(msg.msg.pdepartment);
                $(".personInfomationManageEdit #pposition").val(msg.msg.pposition);
                $(".personInfomationManageEdit #pcreatedate").val(msg.msg.pcreatedate);
                $(".personInfomationManageEdit #personInfomationname").val(msg.msg.pname);
                $(".personInfomationManageEdit #psex").val(msg.msg.psex);
                $(".personInfomationManageEdit #pnation").val(msg.msg.pnation);
                $(".personInfomationManageEdit #pmarriage").val(msg.msg.pmarriage);
                $(".personInfomationManageEdit #ppolitics").val(msg.msg.ppolitics);
                $(".personInfomationManageEdit #pbirthday").val(msg.msg.pbirthday);
                $(".personInfomationManageEdit #pacademic").val(msg.msg.pacademic);
                $(".personInfomationManageEdit #pmajor").val(msg.msg.pmajor);
                $(".personInfomationManageEdit #pschool").val(msg.msg.pschool);
                $(".personInfomationManageEdit #pgraduationdate").val(msg.msg.pgraduationdate);
                $(".personInfomationManageEdit #pdegree").val(msg.msg.pdegree);
                $(".personInfomationManageEdit #pprofessional").val(msg.msg.pprofessional);
                $(".personInfomationManageEdit #ptelephonenumber").val(msg.msg.ptelephonenumber);
                $(".personInfomationManageEdit #phiredate").val(msg.msg.phiredate);
                $(".personInfomationManageEdit #paddress").val(msg.msg.paddress);
                $(".personInfomationManageEdit #pidcard").val(msg.msg.pidcard);
                $(".personInfomationManageEdit #pbankaccount").val(msg.msg.pbankaccount);
                $(".personInfomationManageEdit #popenbank").val(msg.msg.popenbank);
                $(".personInfomationManageEdit #peducationdate1").val(msg.msg.peducationdate1);
                $(".personInfomationManageEdit #peducationdate2").val(msg.msg.peducationdate2);
                $(".personInfomationManageEdit #peducationdate3").val(msg.msg.peducationdate3);
                $(".personInfomationManageEdit #peducationschool1").val(msg.msg.peducationschool1);
                $(".personInfomationManageEdit #peducationschool2").val(msg.msg.peducationschool2);
                $(".personInfomationManageEdit #peducationschool3").val(msg.msg.peducationschool3);
                $(".personInfomationManageEdit #peducationmajor1").val(msg.msg.peducationmajor1);
                $(".personInfomationManageEdit #peducationmajor2").val(msg.msg.peducationmajor2);
                $(".personInfomationManageEdit #peducationmajor3").val(msg.msg.peducationmajor3);
                $(".personInfomationManageEdit #peducationdegree1").val(msg.msg.peducationdegree1);
                $(".personInfomationManageEdit #peducationdegree2").val(msg.msg.peducationdegree2);
                $(".personInfomationManageEdit #peducationdegree3").val(msg.msg.peducationdegree3);
                $(".personInfomationManageEdit #peducationcertificate1").val(msg.msg.peducationcertificate1);
                $(".personInfomationManageEdit #peducationcertificate2").val(msg.msg.peducationcertificate2);
                $(".personInfomationManageEdit #peducationcertificate3").val(msg.msg.peducationcertificate3);
                $(".personInfomationManageEdit #pqualification").val(msg.msg.pqualification);
                $(".personInfomationManageEdit #pqualificationdate").val(msg.msg.pqualificationdate);
                $('.personInfomationManageEdit #pprofessionalcad').attr('checked', msg.msg.pprofessionalcad);
                $('.personInfomationManageEdit #pprofessionalps').attr('checked', msg.msg.pprofessionalps);
                $('.personInfomationManageEdit #pprofessional3dmax').attr('checked', msg.msg.pprofessional3dmax);
                $('.personInfomationManageEdit #pprofessionaltarch').attr('checked', msg.msg.pprofessionaltarch);
                $('.personInfomationManageEdit #pprofessionalother').attr('checked', msg.msg.pprofessionalother);
                $(".personInfomationManageEdit #pexperiencedate1").val(msg.msg.pexperiencedate1);
                $(".personInfomationManageEdit #pexperiencedate2").val(msg.msg.pexperiencedate2);
                $(".personInfomationManageEdit #pexperiencedate3").val(msg.msg.pexperiencedate3);
                $(".personInfomationManageEdit #pexperienceunit1").val(msg.msg.pexperienceunit1);
                $(".personInfomationManageEdit #pexperienceunit2").val(msg.msg.pexperienceunit2);
                $(".personInfomationManageEdit #pexperienceunit3").val(msg.msg.pexperienceunit3);
                $(".personInfomationManageEdit #pexperiencepost1").val(msg.msg.pexperiencepost1);
                $(".personInfomationManageEdit #pexperiencepost2").val(msg.msg.pexperiencepost2);
                $(".personInfomationManageEdit #pexperiencepost3").val(msg.msg.pexperiencepost3);
                $(".personInfomationManageEdit #pexperiencepay1").val(msg.msg.pexperiencepay1);
                $(".personInfomationManageEdit #pexperiencepay2").val(msg.msg.pexperiencepay2);
                $(".personInfomationManageEdit #pexperiencepay3").val(msg.msg.pexperiencepay3);
                $(".personInfomationManageEdit #pexperiencedimission1").val(msg.msg.pexperiencedimission1);
                $(".personInfomationManageEdit #pexperiencedimission2").val(msg.msg.pexperiencedimission2);
                $(".personInfomationManageEdit #pexperiencedimission3").val(msg.msg.pexperiencedimission3);
                $(".personInfomationManageEdit #pexperiencedimissionnumber1").val(msg.msg.pexperiencedimissionnumber1);
                $(".personInfomationManageEdit #pexperiencedimissionnumber2").val(msg.msg.pexperiencedimissionnumber2);
                $(".personInfomationManageEdit #pexperiencedimissionnumber3").val(msg.msg.pexperiencedimissionnumber3);
                $(".personInfomationManageEdit #pfamilyfamilyname1").val(msg.msg.pfamilyfamilyname1);
                $(".personInfomationManageEdit #pfamilyfamilyname2").val(msg.msg.pfamilyfamilyname2);
                $(".personInfomationManageEdit #pfamilyfamilyname3").val(msg.msg.pfamilyfamilyname3);
                $(".personInfomationManageEdit #pfamilyfamilyage1").val(msg.msg.pfamilyfamilyage1);
                $(".personInfomationManageEdit #pfamilyfamilyage2").val(msg.msg.pfamilyfamilyage2);
                $(".personInfomationManageEdit #pfamilyfamilyage3").val(msg.msg.pfamilyfamilyage3);
                $(".personInfomationManageEdit #pfamilyfamilyunit1").val(msg.msg.pfamilyfamilyunit1);
                $(".personInfomationManageEdit #pfamilyfamilyunit2").val(msg.msg.pfamilyfamilyunit2);
                $(".personInfomationManageEdit #pfamilyfamilyunit3").val(msg.msg.pfamilyfamilyunit3);
                $(".personInfomationManageEdit #pfamilyfamilypost1").val(msg.msg.pfamilyfamilypost1);
                $(".personInfomationManageEdit #pfamilyfamilypost2").val(msg.msg.pfamilyfamilypost2);
                $(".personInfomationManageEdit #pfamilyfamilypost3").val(msg.msg.pfamilyfamilypost3);
                $(".personInfomationManageEdit #pfamilyfamilynumber1").val(msg.msg.pfamilyfamilynumber1);
                $(".personInfomationManageEdit #pfamilyfamilynumber2").val(msg.msg.pfamilyfamilynumber2);
                $(".personInfomationManageEdit #pfamilyfamilynumber3").val(msg.msg.pfamilyfamilynumber3);
                $(".personInfomationManageEdit #pemergencycontactname").val(msg.msg.pemergencycontactname);
                $(".personInfomationManageEdit #pemergencycontactnumber").val(msg.msg.pemergencycontactnumber);
                $(".personInfomationManageEdit #pinformationid").attr('checked', msg.msg.pinformationid);
                $(".personInfomationManageEdit #pinformationbank").attr('checked', msg.msg.pinformationbank);
                $(".personInfomationManageEdit #pinformationeducation").attr('checked', msg.msg.pinformationeducation);
                $(".personInfomationManageEdit #pinformationother").attr('checked', msg.msg.pinformationother);
                $("input[name='pemploymentseparation'][id = " + msg.msg.pemploymentseparation + "]").prop("checked", "checked");
                $("input[name='pcriminalrecord'][id = " + msg.msg.pcriminalrecord + "]").prop("checked", "checked");
                $(".personInfomationManageEdit #uid").val(msg.msg.uid);
            }
        }
    });
}
/**
 * 个人档案end
 */

